AE-1470 Clear the pending flag on some toimenpide

muep opened the pull request.

muep committed:
AE-1470 Clear the pending flag on some toimenpide

hkorpi commented on etp-backend/src/main/clj/solita/etp/service/valvonta_oikeellisuus.clj:
```diff
@@ -141,6 +141,12 @@
+                        :energiatodistus
+                        (add-prefix "valvonta$" {:pending false})
+                        ["id = ?" id]
+                        db/default-opts))
```
jos on olemassa oleva palvelu tiedon päivittämiseen niin älä ohita sitä

hkorpi replied:
vai onko jokin syy miksi save-valvonta! ei toimi?

hkorpi replied:
tarkoitus on pitää rajapinta datan muokkaamiseen aina mahdollisimmin pienenä

hkorpi replied:
eli kokeile tuon updaten sijaan:
`(save-valvonta! db id {:pending false})`

muep replied:
Ei ollut erityistä syytä, ei vaan tullut hoksattua tuota vaihtoehtoa. Muutan tuon käyttämään sitä `save-valvonta!`aa.

muep replied:
Pitäisikös muuten olla `(save-valvonta! tx id {:pending false})`, eli `tx` sen `db` sijaan?

hkorpi replied:
joo tuossa pitää olla ehdottomasti tx eli transaktiossa oleva yhteys. Jos tuohon laittaa db niin sitten tämä muutos tehdään transaktion ulkopuolella.

muep replied:
Nämä on nyt korjailtu

muep committed:
AE-1470 Use tx instead of db, call save-valvonta!

The previous version was unnecessarily duplicating functionality from
save-valvonta!

More severely, it was passing db instead of tx to the jdbc layer. This
would do the change to non-pending state outside the transaction that
we are supposed to use.
